### ISSUE TITLE
Implement support for QueryStrings

### ISSUE THREAD

**Issue by aalises** [k1r0s/preact-routlet#8]
Currently `preact-routlet` does not support **querystrings** on the URL (e.g `/testpath?foo=bar&zzz=baz`). When adding them to a url, the router does not recognize the path.
 
It would be nice to implement recognizing querystrings on the routlet by adding another prop, `search`, which includes the querystring in the format `this.props.search = "?param1=value1&param2=value2"`

**Comment by aalises** [comment]
Made a pull request adding support for qs: #9

**Comment by k1r0s** [comment]
fair point, but why specifically __querystring__ format?

ie: I have a component that listens to /user-list/:searchParams, so it receives params like this `/user-list/name,jose` otherwise `/user-list/null`

**Comment by aalises** [comment]
@k1r0s yes! We already use the path parameter parsing, they're amazing. The support for query strings is just adding more flexibility to the parsing by adding minimal change.

In our case, the backend links sometimes return URLs with query string parameters, so in-path parameters are also used within the application but we also need a way to obtain these query string parameters.
